lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2590,8 +2590,39 @@ namespace {
     }
 
     Decl *VisitNamespaceAliasDecl(const clang::NamespaceAliasDecl *decl) {
-      // FIXME: Implement once Swift has namespaces.
-      return nullptr;
+      Optional<ImportedName> correctSwiftName;
+      auto importedName = importFullName(decl, correctSwiftName);
+      auto name = importedName.getDeclName().getBaseIdentifier();
+      if (name.empty())
+        return nullptr;
+
+      if (correctSwiftName)
+        return importCompatibilityTypeAlias(decl, importedName,
+                                            *correctSwiftName);
+
+      auto dc =
+          Impl.importDeclContextOf(decl, importedName.getEffectiveContext());
+      if (!dc)
+        return nullptr;
+
+      auto *aliasedDecl = dyn_cast<TypeDecl>(
+          Impl.importDecl(decl->getAliasedNamespace(),
+                          getActiveSwiftVersion()));
+      if (!aliasedDecl)
+        return nullptr;
+
+      auto result = Impl.createDeclWithClangNode<TypeAliasDecl>(
+          decl,
+          AccessLevel::Public,
+          Impl.importSourceLoc(decl->getBeginLoc()),
+          SourceLoc(),
+          name,
+          Impl.importSourceLoc(decl->getLocation()),
+          /*GenericParams=*/ nullptr,
+          dc);
```

Comment:
Nit: Was this code formatted using `clang-format`? It seems a bit unusual that the spacing would be spread out like so, most other code is not formatted like this.

Reply:
That was a formatter misconfiguration on my side, thanks for noticing! Fixed.